src/main/ts/components/Editor.tsx

```diff
@@ -8,21 +8,30 @@ import { Bookmark, Editor as TinyMCEEditor, EditorEvent, TinyMCE } from 'tinymce
 
 type EditorOptions = Parameters<TinyMCE['init']>[0];
 
-export interface IProps {
+export type Version = `${'4' | '5' | '6' | '7'}${'' | '-dev' | '-testing' | `.${number}` | `.${number}.${number}`}`;
+
+export interface CloudHostedProps {
   apiKey: string;
+}
+
+export interface HybridOrSelfHostedProps {
+  tinymceScriptSrc: string | string[] | ScriptItem[];
```

Comment:
I don't think this should be required. Since you might be loading tinymce though some other method like in the head element or though some other component.

Reply:
Moved it back to `IProps`, so it's optional again